src/Shared/CommunicationsUtilities.cs

```diff
@@ -110,8 +115,13 @@ protected internal Handshake(HandshakeOptions nodeType)
             fileVersionMinor = fileVersion.Minor;
             fileVersionBuild = fileVersion.Build;
             fileVersionPrivate = fileVersion.Revision;
-            using Process currentProcess = Process.GetCurrentProcess();
-            sessionId = currentProcess.SessionId;
+
+            // This reaches out to NtQuerySystemInformation. Due to latency, allow skipping for derived handshake if unused.
+            if (includeSessionId)
```

Comment:
Re-arranged a bit here as well. Found this also calling into `GetCurrentProcess` + `NtQuerySystemInformation` on every build, which isn't used at all by `ServerNodeHandshake`.